apps/frontend/components/pages/Report/ReportContainer.tsx

```diff
@@ -34,6 +35,10 @@ export const ReportContainer = () => {
 
   const { data: formsData, isLoading } = useGetForms(searchReportData);
 
+  const filteredFormsData = useMemo(() => {
+    return formsData !== undefined ? dropNotApproved(formsData) : undefined;
```

Comment:
Alternatively, dropNotApproved could accept undefined values and return undefined early